homeassistant/components/template/number.py

```diff
@@ -0,0 +1,241 @@
+        vol.Required(CONF_ATTRIBUTES): vol.Schema(
+            {
+                vol.Required(ATTR_STEP): cv.template,
+                vol.Optional(ATTR_MIN, default=DEFAULT_MIN_VALUE): cv.template,
+                vol.Optional(ATTR_MAX, default=DEFAULT_MAX_VALUE): cv.template,
+            }
+        ),
+        vol.Optional(CONF_AVAILABILITY): cv.template,
+        vol.Optional(CONF_OPTIMISTIC, default=DEFAULT_OPTIMISTIC): cv.boolean,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
+    }
+)
+
+
+async def _async_create_entities(
+    hass: HomeAssistant, definitions: list[dict[str, Any]], unique_id_prefix: str | None
+) -> list[TemplateNumber]:
+    """Create the Template number."""
+    entities = []
+    for definition in definitions:
+        unique_id = definition.get(CONF_UNIQUE_ID)
+        if unique_id and unique_id_prefix:
+            unique_id = f"{unique_id_prefix}-{unique_id}"
+        entities.append(
+            TemplateNumber(
+                hass,
+                definition[CONF_NAME],
+                definition[CONF_STATE],
+                definition.get(CONF_AVAILABILITY),
+                definition[CONF_SET_VALUE],
+                definition[CONF_ATTRIBUTES][ATTR_STEP],
+                definition[CONF_ATTRIBUTES][ATTR_MIN],
+                definition[CONF_ATTRIBUTES][ATTR_MAX],
+                definition[CONF_OPTIMISTIC],
+                unique_id,
+            )
+        )
+    return entities
+
+
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: Config,
+    async_add_entities: AddEntitiesCallback,
+    discovery_info: dict[str, Any] | None = None,
+) -> None:
+    """Set up the template number."""
+    if "coordinator" in discovery_info:
+        async_add_entities(
+            TriggerNumberEntity(hass, discovery_info["coordinator"], config)
+            for config in discovery_info["entities"]
+        )
+        return
+
+    async_add_entities(
+        await _async_create_entities(
+            hass, discovery_info["entities"], discovery_info["unique_id"]
+        )
+    )
+
+
+class TemplateNumber(TemplateEntity, NumberEntity):
+    """Representation of a template number."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        name_template: Template,
+        value_template: Template,
+        availability_template: Template | None,
+        command_set_value: dict[str, Any],
+        step_template: Template,
+        minimum_template: Template | None,
+        maximum_template: Template | None,
+        optimistic: bool,
+        unique_id: str | None,
+    ) -> None:
+        """Initialize the number."""
+        super().__init__(availability_template=availability_template)
+        self._attr_name = DEFAULT_NAME
+        self._name_template = name_template
+        name_template.hass = hass
+        try:
+            self._attr_name = name_template.async_render(parse_result=False)
+        except TemplateError:
+            pass
```

Comment:
Could use `contextlib.supress` here

Reply:
good tip! This seems to be the pattern used in other platforms for this integration, so I am thinking maybe we have a separate PR to do this across the board?

Reply:
Agreed that the others should be done in a new PR. Should be ok to do this one now since its new

Reply:
Fair enough 👍🏾